### PR TITLE
OnScreenDisplay: Allow for different colored messages

### PR DESCRIPTION
**Pull request by lioncash**

### CODE REVIEW

File: Source/Core/VideoCommon/OnScreenDisplay.cpp [hunk 1]
```diff
@@ -21,18 +21,23 @@ namespace OSD
 struct Message
 {
 	Message() {}
-	Message(const std::string& s, u32 ts) : str(s), timestamp(ts) {}
+	Message(const std::string& s, u32 ts, u32 rgb) : m_str(s), m_timestamp(ts), m_rgb(rgb)
+	{
+		if (m_rgb > 0xFFFFFF)
+			m_rgb = 0xFFFFFF;
```
Comment (degasus) [inline Source/Core/VideoCommon/OnScreenDisplay.cpp]: I would prefer masking.

File: Source/Core/VideoCommon/OnScreenDisplay.cpp [hunk 2]
```diff
@@ -56,7 +61,7 @@ void DrawMessages()
 
 		alpha <<= 24;
 
-		g_renderer->RenderText(it->str, left, top, 0xffff30 | alpha);
+		g_renderer->RenderText(it->m_str, left, top, it->m_rgb | alpha);
```
Comment (degasus) [inline Source/Core/VideoCommon/OnScreenDisplay.cpp]: Fancy feature: Let's move the masking here and multiply the alpha value, so it's allowed to set an alpha value per message.

Comment (lioncash) [inline Source/Core/VideoCommon/OnScreenDisplay.cpp]: What do you mean by multiply the alpha?

Comment (degasus) [inline Source/Core/VideoCommon/OnScreenDisplay.cpp]: (it->m_rgb & 0xffffff) | ((it->m_rgb >> 24) * alpha)

and alpha goes from 1.0 to 0.0

**Comment by Buddybenj** [comment]
Bump.

**Comment by lioncash** [comment]
@degasus Ok, so I moved your suggestions into place. All that's left is how the alpha is handled. How do you want me to change the `alpha` variable?

File: Source/Core/VideoCommon/OnScreenDisplay.h [hunk 3]
```diff
@@ -11,8 +11,8 @@
 
 namespace OSD
 {
-// On-screen message display
-void AddMessage(const std::string& str, u32 ms = 2000);
+// On-screen message display (colored yellow by default)
+void AddMessage(const std::string& str, u32 ms = 2000, u32 rgb = 0xFFFF30);
```
Comment (degasus) [inline Source/Core/VideoCommon/OnScreenDisplay.h]: I guess default value for rgb should be 0xFFFFFF30

File: Source/Core/VideoCommon/OnScreenDisplay.cpp [hunk 4]
```diff
@@ -56,7 +59,7 @@ void DrawMessages()
 
 		alpha <<= 24;
 
-		g_renderer->RenderText(it->str, left, top, 0xffff30 | alpha);
+		g_renderer->RenderText(it->m_str, left, top, (it->m_rgb & 0xFFFFFF) | ((it->m_rgb >> 24) * alpha));
```
Comment (degasus) [inline Source/Core/VideoCommon/OnScreenDisplay.cpp]: float alpha = max(1.0, min(0.0, time_left / 1024.0))
u32 color = it->m_rgb & 0xFFFFFF;
color |= (u32)((it->m_rgb >> 24) * alpha) << 24;
g_renderer->RenderText(it->m_str, left, top, color);

**Comment by degasus** [comment]
LGTM

thanks for keeping care about my superfluous ideas

**Comment by lioncash** [comment]
@degasus Can you look this over one last time, I mostly just changed the `rgb` and `m_rgb` vars into `rgba` and `m_rgba`

**Pull request merged**